chisel/src/main.rs

```diff
@@ -269,6 +277,9 @@ fn execute_module(context: &ModuleContext, module: &mut Module) -> bool {
 fn chisel_execute(context: &ChiselContext) -> Result<bool, &'static str> {
     if let Ok(buffer) = read(context.file()) {
         if let Ok(mut module) = deserialize_buffer::<Module>(&buffer) {
+            // FIXME: this parsing is needed for binaryenopt
+            // Should this be run depending on binaryenopt is requested or not?
```

Comment:
Ideally yes, this can be done by just adding a new method querying the context on whether the module is requested.

Reply:
Actually I think it is useful to do, because other filters may depend on it too.